src/_path.h

```diff
@@ -872,6 +872,64 @@ bool path_intersects_path(PathIterator1 &p1, PathIterator2 &p2)
     return false;
 }
 
+// returns whether the segment from (x1,y1) to (x2,y2)
+// intersects the rectangle centered at (cx,cy) with size (w,h)
+inline bool segment_intersects_rectangle(double x1, double y1,
```

Comment:
It's difficult to explain with text alone. Is there a place where I can put an SVG or something to illustrate it?